lib/rspec/matchers/built_in/has.rb

```diff
@@ -34,25 +36,25 @@ def does_not_match?(actual, &block)
         # @api private
         # @return [String]
         def failure_message
-          validity_message || "expected ##{predicate}#{failure_message_args_description} to return true, got false"
+          failure_message_expecting(true)
         end
 
         # @api private
         # @return [String]
         def failure_message_when_negated
-          validity_message || "expected ##{predicate}#{failure_message_args_description} to return false, got true"
+          failure_message_expecting(false)
         end
 
         # @api private
         # @return [String]
         def description
-          [method_description, args_description].compact.join(' ')
+          "#{method_description}#{args_to_sentence}"
         end
 
       private
 
         def predicate_accessible?
-          !private_predicate? && predicate_exists?
+          @actual.respond_to? predicate
         end
```

Comment:
This change is fine for be, but changes has behaviour?

Reply:
Ahh actually I see, its not used other than in validity message by `Has, but is used by `Be`.